### PR TITLE
Issue #SB-2123 feat: Org search and user search

### PR DESCRIPTION
**Pull request by souravdey091**

### CODE REVIEW

File: src/app/client/src/app/modules/core/components/search/search.component.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
-import { Component } from '@angular/core';
+import { SearchService } from './../../services/search/search.service';
```
Comment (anoophm) [inline src/app/client/src/app/modules/core/components/search/search.component.ts]: ''./../../services'

File: src/app/client/src/app/app.module.ts [hunk 2]
```diff
@@ -17,6 +17,7 @@ import { LearnModule } from '@sunbird/learn';
 import { WorkspaceModule } from '@sunbird/workspace';
 import { OrgManagementModule } from '@sunbird/org-management';
 import { PublicModule } from '@sunbird/public';
+import { SearchModule } from './modules/search/search.module';
```
Comment (sunilpes) [inline src/app/client/src/app/app.module.ts]: Create module Alias like @sunbird/search

File: src/app/client/src/app/modules/core/components/search/search.component.ts [hunk 3]
```diff
@@ -7,9 +12,101 @@ import { Component } from '@angular/core';
...
+      queryParams: this.queryParam
+    });
+  }
+  /**
+   * gets the current url,
+   * and queryParams
+   */
+  ngOnInit() {
+
+    this.search = this.config.dropDownConfig.FILTER.SEARCH.search;
```
Comment (sunilpes) [inline src/app/client/src/app/modules/core/components/search/search.component.ts]: put it inside functions

File: src/app/client/src/app/modules/search/components/user-delete/user-delete.component.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      }
+    );
+  }
+
+  /**
+   * This method helps to redirect to the parent component
+   * page, i.e, outbox listing page with proper page number
+	 *
+	 */
+  redirect(): void {
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-delete/user-delete.component.ts]: change to redirectToParent

File: src/app/client/src/app/modules/search/components/user-delete/user-delete.component.ts [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    this.routerNavigationService.navigateToParentUrl(this.activatedRoute.snapshot);
+  }
+
+  /**
+   * This method sets the annmouncementId and pagenumber from
+   * activated route
+	 */
+  ngOnInit() {
+    this.activatedRoute.params.subscribe(params => {
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-delete/user-delete.component.ts]: Move this code inside a function

File: src/app/client/src/app/modules/core/components/search/search.component.spec.ts [hunk 6]
```diff
@@ -1,26 +1,53 @@
...
-
   beforeEach(() => {
+    router = TestBed.get(Router);
+    location = TestBed.get(Location);
     fixture = TestBed.createComponent(SearchComponent);
     component = fixture.componentInstance;
-    fixture.detectChanges();
   });
-
   it('should create', () => {
```
Comment (sunilpes) [inline src/app/client/src/app/modules/core/components/search/search.component.spec.ts]: redundant test case

File: src/app/client/src/app/modules/core/components/search/search.component.ts [hunk 7]
```diff
@@ -7,9 +12,101 @@ import { Component } from '@angular/core';
...
+    this.searchUrl = this.config.dropDownConfig.FILTER.SEARCH.searchUrl;
+    this.activatedRoute.queryParams.subscribe(queryParams => {
+      this.queryParam = { ...queryParams };
+      this.key = this.queryParam['key'];
+    });
+    this.route.events
+      .filter(e => e instanceof NavigationEnd).subscribe((params: any) => {
+        console.log(params);
```
Comment (sunilpes) [inline src/app/client/src/app/modules/core/components/search/search.component.ts]: remove console logs

File: src/app/client/src/app/modules/search/components/org-filter/org-filter.component.ts [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  resetFilters() {
+    this.refresh = false;
+    this.queryParams = {};
+    this.router.navigate(['/search/Organisations', 1]);
+    setTimeout(() => {
+      this.refresh = true;
+    }, 0);
+  }
+  ngOnInit() {
+    _.forIn(this.queryParams, (value, key) => {
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/org-filter/org-filter.component.ts]: push it to some functions

File: src/app/client/src/app/modules/search/components/org-search/org-search.component.spec.ts [hunk 9]
```diff
@@ -0,0 +1,90 @@
...
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(OrgSearchComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/org-search/org-search.component.spec.ts]: redundant test cases

File: src/app/client/src/app/modules/search/components/org-search/org-search.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,90 @@
...
+    spyOn(searchService, 'orgSearch').and.callFake(() => Observable.of(Response.successData));
+    component.populateOrgSearch();
+    fixture.detectChanges();
+    expect(component.searchList).toBeDefined();
+    expect(component.totalCount).toBeDefined();
+    expect(component.showLoader).toBeFalsy();
+    expect(component.noResult).toBeFalsy();
+  });
+
+  it('should throw error when searchService api is not called', () => {
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/org-search/org-search.component.spec.ts]: please add proper description

File: src/app/client/src/app/modules/search/components/org-search/org-search.component.ts [hunk 11]
```diff
@@ -0,0 +1,218 @@
...
+          this.noResultMessage = {
+            'message': this.resourceService.messages.stmsg.m0008,
+            'messageText': this.resourceService.messages.stmsg.m0007
+          };
+        }
+      },
+      err => {
+        this.showLoader = false;
+        this.noResult = false;
+        // this.toasterService.error(this.resourceService.messages.fmsg.m0006);
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/org-search/org-search.component.ts]: remove unused code

File: src/app/client/src/app/modules/search/components/org-search/org-search.component.ts [hunk 12]
```diff
@@ -0,0 +1,218 @@
...
+      'orgName': 'Organisation Name'
+    }];
+
+    _.each(this.searchList, (key, index) => {
+      downloadArray.push({
+        'orgName': key.orgName
+      });
+    });
+
+    const csv = new Angular2Csv(downloadArray, 'Organisations', options);
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/org-search/org-search.component.ts]: return new object, unused const value

File: src/app/client/src/app/modules/search/components/org-search/org-search.component.ts [hunk 13]
```diff
@@ -0,0 +1,218 @@
...
+    _.forIn(event, (value, key) => {
+      if (value.length > 0) {
+        queryParams[key] = value;
+      }
+    });
+    this.route.navigate(['/search/Organisations', 1], { queryParams: queryParams });
+  }
+
+  ngOnInit() {
+    Observable
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/org-search/org-search.component.ts]: push this to separate function

File: src/app/client/src/app/modules/search/components/user-delete/user-delete.component.spec.ts [hunk 14]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-delete/user-delete.component.spec.ts]: no test cases???

File: src/app/client/src/app/modules/search/components/user-delete/user-delete.component.ts [hunk 15]
```diff
@@ -0,0 +1,142 @@
...
+export class UserDeleteComponent implements OnInit {
+  /**
+	 * Contains unique announcement id
+	 */
+  userId: string;
+
+  /**
+	 * Contains page number of outbox list
+	 */
+  pageNumber = 1;
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-delete/user-delete.component.ts]: add type number

File: src/app/client/src/app/modules/search/components/user-edit/user-edit.component.spec.ts [hunk 16]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-edit/user-edit.component.spec.ts]: no proper test cases

File: src/app/client/src/app/modules/search/components/user-edit/user-edit.component.ts [hunk 17]
```diff
@@ -0,0 +1,236 @@
...
+  selectedOrgUserRoles: Array<string>;
+  selectedOrgUserRolesNew: any;
+
+  /**
+	 * Contains announcement details returned from API or object called from
+   * announcement service
+	 */
+  userDetails: any;
+
+  allRoles = [
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-edit/user-edit.component.ts]: get it from API

File: src/app/client/src/app/modules/search/components/user-edit/user-edit.component.ts [hunk 18]
```diff
@@ -0,0 +1,236 @@
...
+        _.each(this.userDetails.organisations, (org) => {
+          const orgNameAndId = _.find(orgApiResponse.result.response.content, function (organisation) {
+            return organisation.id === org.organisationId;
+          });
+          if (orgNameAndId) { org.orgName = orgNameAndId.orgName; }
+        });
+      }
+    );
+
+    console.log('this.userDetails1-----', this.userDetails);
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-edit/user-edit.component.ts]: remove this line

File: src/app/client/src/app/modules/search/components/user-filter/user-filter.component.spec.ts [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+    fixture = TestBed.createComponent(UserFilterComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  it('should call resetFilters method ', inject([ConfigService, Router],
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-filter/user-filter.component.spec.ts]: describe "it" properly, remove duplicate description in below test cases

File: src/app/client/src/app/modules/search/components/user-profile/user-profile.component.spec.ts [hunk 20]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/components/user-profile/user-profile.component.spec.ts]: no test cases???

File: src/app/client/src/app/modules/search/services/user-search/user-search.service.spec.ts [hunk 21]
```diff
@@ -0,0 +1,15 @@
+import { TestBed, inject } from '@angular/core/testing';
```
Comment (sunilpes) [inline src/app/client/src/app/modules/search/services/user-search/user-search.service.spec.ts]: no test cases??

**Pull request merged**